### ISSUE TITLE
HPX 0.11.1 memory error when scanning large directory 

### ISSUE THREAD

**Issue by waz1500** [happypandax/happypandax#150]
**Describe the bug**
Scanning a directory with over 7000 folders+ results in the server crashing. 

**To Reproduce**
1. Have a large collection
2. Click on Scan and add. All options are off including nested folders.
3. Folders are detected. Click on Add
4. HPX freezes and displays error
`[CommandError|120]: Command(121) has failed with: [HappypandaError|100]: MemoryError()`

This occurs with the Desktop version of HPX also and because they both crash at different stages of the process has it's own issue here. https://github.com/happypandax/desktop/issues/2

Using the standalone server and browser I can at least get to step 3. Using the Desktop version, and it will freeze on step 2. 

**Expected behavior**
Directory should scan. Previous versions of HPX managed to handle this folder size before without too much problem. 

**Environment:**
 - OS: Windows 10
 - Version 76.0.3809.132 (Official Build) (64-bit)
 - HPX Version 0.11.1

**Comment by twiddli** [comment]
> Previous versions of HPX managed to handle this folder size before without too much problem.

So this doesn't occur with `0.10.0`?

**Comment by waz1500** [comment]
Confirmed twiddli.

Adding the exact same directory works in the `0.10.2.win` release. The releases after ` 0.11.0.` and ` 0.11.1.` won't work with my collection.

`0.10.2.win` release.

![hpx chg](https://user-images.githubusercontent.com/17833884/64082625-c5e48b80-cd09-11e9-8779-ef427220dce9.jpg)
The progress bar moves, giving a clear indication on added manga. 
________

` 0.11.0.` and ` 0.11.1.` releases 
Clicking "add" causes the client to display the "completed" text as seen below. The progress bar remains at 0.
![strange](https://user-images.githubusercontent.com/17833884/64082626-c9781280-cd09-11e9-82ae-bd39f3c83474.jpg)

This is what appears in the server afterwards. 
![strange2](https://user-images.githubusercontent.com/17833884/64082627-cda43000-cd09-11e9-9a83-3a8d9ac5f5e8.jpg)

I've had to zip the debug file since it becomes a whopping 98mb file on a fresh install. 
[debug.zip](https://github.com/happypandax/happypandax/files/3563936/debug.zip)

Hopefully this can help you diagnose what broke in the  0.11.x.` release with fairly largish collections.
